modules/Movecraft/src/main/java/net/countercraft/movecraft/listener/BlockListener.java

```diff
@@ -387,9 +387,12 @@ private void processAssault(EntityExplodeEvent e){
 
                 // whether or not you actually destroyed the block, add to damages
                 long damages = assault.getDamages() + Settings.AssaultDamagesPerBlock;
-                //if (damages < assault.getMaxDamages()) {
+                if (damages < assault.getMaxDamages()) {
```

Comment:
There is still an edge case that can cause the damages to exceed the maximum. Instead, just pass the minimum of `damages` and `assault::getMaxDamages` to `assault::setDamages`. This allows the removal of this if else clause and avoids any possible overflow.